js/src/data/Projection.ts

```diff
@@ -0,0 +1,34 @@
+
+
+// Intended to match the results of
+// https://github.com/mapbox/vector-tile-js/blob/77851380b63b07fd0af3d5a3f144cc86fb39fdd1/lib/vectortilefeature.js#L129
+
+export class Projection {
+    size: number;
+    x0: number;
+    y0: number;
+    s1: number;
+    s2: number;
+
+    constructor(x: number, y: number, z: number) {
+        const extent = 4096;
```

Comment:
shouldn't this be per layer rather than hardcoded?

Reply:
Nice catch! I was meaning to fix this, but had forgotten. Solved in 100644

Reply:
/cc @ebrelsford `feature.extent` is now available.